Panic  during follow

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb29e8f]
goroutine 72 [running]:
github.com/drand/drand/core.(*Drand).StartFollowChain.func1(0xc0002240c0)
        /home/runner/work/drand/drand/core/drand_control.go:801 +0x4f
github.com/drand/drand/core.(*Drand).StartFollowChain(0xc0002240c0, 0xc00034fda0, 0xe7f940, 0xc00038c530, 0xe61f40, 0xc000015290)
        /home/runner/work/drand/drand/core/drand_control.go:848 +0xe33
github.com/drand/drand/protobuf/drand._Control_StartFollowChain_Handler(0xd32de0, 0xc0002240c0, 0xe7d0c0, 0xc000224300, 0x1484500, 0xc0002c0600)
        /home/runner/work/drand/drand/protobuf/drand/control_grpc.pb.go:397 +0x109
google.golang.org/grpc.(*Server).processStreamingRPC(0xc00026e820, 0xe810e0, 0xc000226d80, 0xc0002c0600, 0xc000352e70, 0x143e820, 0x0, 0x0, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:1329 +0xcbd
google.golang.org/grpc.(*Server).handleStream(0xc00026e820, 0xe810e0, 0xc000226d80, 0xc0002c0600, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:1409 +0xc66
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc00020b790, 0xc00026e820, 0xe810e0, 0xc000226d80, 0xc0002c0600)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:746 +0xa1
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:744 +0xa1
```

--- a/core/drand_control.go
+++ b/core/drand_control.go
@@ -800,7 +800,7 @@ func (d *Drand) StartFollowChain(req *drand.StartFollowRequest, stream drand.Con
 		d.state.Lock()
 		if d.syncerCancel != nil {
 			// it can be nil when we recreate a new beacon we cancel it
-			// see drand.go
+			// see drand.go:newBeacon()
 			d.syncerCancel()
 		}
 		d.syncerCancel = nil

--- a/core/drand_control.go
+++ b/core/drand_control.go
@@ -798,7 +798,11 @@ func (d *Drand) StartFollowChain(req *drand.StartFollowRequest, stream drand.Con
 	d.state.Unlock()
 	defer func() {
 		d.state.Lock()
-		d.syncerCancel()
+		if d.syncerCancel != nil {
+			// it can be nil when we recreate a new beacon we cancel it
+			// see drand.go
+			d.syncerCancel()
+		}
 		d.syncerCancel = nil
 		d.state.Unlock()
 	}()

